controllers/azuremanagedmachinepool_reconciler.go

```diff
@@ -74,22 +75,32 @@ func (a *AgentPoolVMSSNotFoundError) Is(target error) bool {
 
 // newAzureManagedMachinePoolService populates all the services based on input scope.
 func newAzureManagedMachinePoolService(scope *scope.ManagedMachinePoolScope) (*azureManagedMachinePoolService, error) {
-	var authorizer azure.Authorizer = scope
-	if scope.Location() != "" {
-		regionalAuthorizer, err := azure.WithRegionalBaseURI(scope, scope.Location())
-		if err != nil {
-			return nil, errors.Wrap(err, "failed to create a regional authorizer")
-		}
-		authorizer = regionalAuthorizer
+	scaleSetAuthorizer, err := scaleSetAuthorizer(scope)
+	if err != nil {
+		return nil, err
 	}
 
 	return &azureManagedMachinePoolService{
 		scope:         scope,
 		agentPoolsSvc: agentpools.New(scope),
-		scaleSetsSvc:  scalesets.NewClient(authorizer),
+		scaleSetsSvc:  scalesets.NewClient(scaleSetAuthorizer),
 	}, nil
 }
 
+// scaleSetAuthorizer takes a scope and determines if a regional authorizer is needed for scale sets
+// see https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/1850 for context on region based authorizer.
+func scaleSetAuthorizer(scope *scope.ManagedMachinePoolScope) (azure.Authorizer, error) {
+	if scope.Location() == "" {
+		return scope, nil // no location so use default
+	}
+
+	if scope.ControlPlane.Spec.AzureEnvironment == azureautorest.USGovernmentCloud.Name {
```

Comment:
unfortunately, i dont have access to any of those clouds. if there is anyone internal at msft who could confirm I would gladly add additional code to skip more clouds if we need too

Reply:
yup, i was trying to create an AKS cluster in usgovarizona and it was failing DNS trying to contact the following:
```
https://usgovarizona.management.usgovcloudapi.net/subscriptions/5129b42f-3fc6-48e2-b674-eebe99ca6121/resourceGroups/SPC_MC_rnl-rg_luther-test_usgovarizona/providers/Microsoft.Compute/virtualMachineScaleSets?api-version=2021-11-01
```

and DNS fails for all the regions I checked for <region>.management.usgovcloudapi.net and after adding this code I retested and an AKS cluster was created successfully

Reply:
i dont know if it hit the status. the dns errors were from capz logs, i didn't check the status tbh

Reply:
I think if an error show up in at least one of those places that's reasonably covered.

Reply:
In this case I think it would be better to exclude all clouds by default and only opt-in when we know for sure that regional URIs are supported. The case where "we tried to use a regional base URI but it wasn't supported" is worse (total failure) than "we didn't use a regional base URI when we could have" (de-optimization which may cause transient failures)

```suggestion
	if scope.ControlPlane.Spec.AzureEnvironment != azure.PublicCloudName {
		return scope, nil // no region support in sovereign clouds
	}
```

Note that I intentionally used the AzurePublicCloudName const in https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/azure/defaults.go#L34 to avoid adding anoter usage of `azureautorest` since it's deprecated.

Reply:
@CecileRobertMichon so something more like this?

```
func scaleSetAuthorizer(scope *scope.ManagedMachinePoolScope) (azure.Authorizer, error) {
	if scope.Location() == "" {
		return scope, nil // no location so use default
	}

	if scope.ControlPlane.Spec.AzureEnvironment == azure.PublicCloudName {
		return azure.WithRegionalBaseURI(scope, scope.Location()) // no region support in usgovcloud
	}

	return scope, nil
}
```

Reply:
i dont know why we do the location check in this method, so probably drop that too i guess

Reply:
maybe this is enough...
```
func scaleSetAuthorizer(scope *scope.ManagedMachinePoolScope) (azure.Authorizer, error) {
	if scope.ControlPlane.Spec.AzureEnvironment == azure.PublicCloudName {
		return azure.WithRegionalBaseURI(scope, scope.Location()) // public cloud supports regional end points
	}

	return scope, nil
}
```

Reply:
committed